test/api/test_subscription_event.py

```diff
@@ -0,0 +1,238 @@
+            request_headers={'Authorization': 'Basic dG9rZW46'},
+            status_code=204,
+            json = data
+        )
+
+        config = Config("token")  # is actually checked in mock
+        result = SubscriptionEvent.destroy_modify_with_params(config, data = data).get()
+
+        self.assertEqual(mock_requests.call_count, 1, "expected call")
+        self.assertEqual(mock_requests.last_request.qs, {})
+        self.assertTrue(result is None)
+
+    @requests_mock.mock()
+    def test_modify_subscription_event_with_id(self, mock_requests):
+        data={
+            "id": 7654321,
+            "amount_in_cents": 10
+        }
+        mock_requests.register_uri(
+            'PATCH',
+            "https://api.chartmogul.com/v1/subscription_events",
+            request_headers={'Authorization': 'Basic dG9rZW46'},
+            status_code=200,
+            json=expected_sub_ev
+        )
+        config = Config("token")  # is actually checked in mock
+        sub_ev = SubscriptionEvent.modify_with_params(config, data = data).get()
+
+        expected = SubscriptionEvent(**expected_sub_ev)
+        self.assertEqual(mock_requests.call_count, 1, "expected call")
+        self.assertEqual(mock_requests.last_request.qs, {})
+        self.assertEqual(mock_requests.last_request.json(), data)
+        self.assertTrue(isinstance(sub_ev, SubscriptionEvent))
+        self.assertEqual(sub_ev.id, 7654321)
+
+    @requests_mock.mock()
+    def test_modify_subscription_event_with_ds_uuid_and_external_id(self, mock_requests):
+        data={
+            "external_id": "evnt_026",
+            "data_source_uuid": "ds_1fm3eaac-62d0-31ec-clf4-4bf0mbe81aba",
+            "amount_in_cents": 10
+        }
+        mock_requests.register_uri(
+            'PATCH',
+            "https://api.chartmogul.com/v1/subscription_events",
+            request_headers={'Authorization': 'Basic dG9rZW46'},
+            status_code=200,
+            json=expected_sub_ev
+        )
+        config = Config("token")  # is actually checked in mock
+        sub_ev = SubscriptionEvent.modify_with_params(config, data=data).get()
+
+        expected = SubscriptionEvent(**expected_sub_ev)
+        self.assertEqual(mock_requests.call_count, 1, "expected call")
+        self.assertEqual(mock_requests.last_request.qs, {})
+        self.assertEqual(mock_requests.last_request.json(),data)
+        self.assertTrue(isinstance(sub_ev, SubscriptionEvent))
+        self.assertEqual(sub_ev.id, 7654321)
+
+    @requests_mock.mock()
+    def test_modify_subscription_event_with_bad_params(self, mock_requests):
+        data={
+            "external_id": "evnt_026",
+            "amount_in_cents": 10
+        }
+        mock_requests.register_uri(
+            'PATCH',
+            "https://api.chartmogul.com/v1/subscription_events",
+            request_headers={'Authorization': 'Basic dG9rZW46'},
+            status_code=200,
```

Comment:
Are you sure the status code will be 200?

Reply:
This will be 400 Error.